src/wallet/mod.rs

```diff
@@ -983,6 +983,59 @@ impl Wallet {
         tracing::trace!("Trusted pending: {}", balance.trusted_pending);
         Ok(balance)
     }
+ 
+    pub async fn list_wallet_transactions(&self) -> Result<Vec<WalletTransaction>> {
+        let wallet = self.bitcoin_wallet.lock();
+        let transactions = wallet.transactions();
+        let mut txs = Vec::new();
+        
+        for tx in transactions {
+            // Calculate total input value
+            let mut input_value = 0;
+            for input in tx.tx_node.tx.input.iter() {
+                if let Ok(prev_output) = self.bitcoin_blockchain.fetch_tx(input.previous_output.txid) {
+                    input_value += prev_output.output[input.previous_output.vout as usize].value.to_sat();
+                }
+            }
```

Comment:
This seems ... suboptimal. If we're fetching TXs from elsewhere we should instead use `getrawtransaction` from Bitcoin Core IMO, and avoid doing a RPC call _for each input_ in the TX. But do we really need to fetch anything from elsewhere?

Reply:
I don't know how else to do it. I don't get enough information from bdk.list_transactions. Only get general info of the prev output, but no amounts..

Reply:
It is possible to batch JSON-RPC requests, so you could batch-request `getrawtransaction`

Reply:
I swapped to using getrawtransaction now